src/components/SumField.tsx

```diff
@@ -31,33 +51,40 @@ class SumFields extends React.Component<any, any> {
                 fontStyle={FontStyle.NORMAL}
                 marginTop={MarginTop.SMALL}
               >
-                Cost per hour: { this.computeSalarySum() }
+                Cost per hour: { this.renderSalarySum() }
               </Text>
               
               
           </Col>
       )
   }
+
+  renderSalarySum = () => {
+    this.computeSalarySum();
```

Comment:
This is very un-React-like. It's dangerous to update props in a render function, as it could lead to infinite loops. The React-way is to always make the ui adapt to changes in the state, and render should be a [pure function](https://www.sitepoint.com/functional-programming-pure-functions/).

I would suggest moving the `computeSalarySum` call inside the interval function, or even better, as salary sum really is a [computed value](https://mobx.js.org/refguide/computed-decorator.html) it needn't be stored in the state at all :)